### ISSUE TITLE
[Bug]: Inconsistency in focus behavior between Combo box and Dropdown

### ISSUE THREAD

**Issue by Kritvi-bhatia17** [carbon-design-system/carbon#16106]
## Acceptance criteria
- [x] Design needs to do some research on what the best practice is of whether to cycle through dropdown items with focus in a menu.
 - Should this interaction be the same for Dropdown, Combobox, Multiselect, and Select?
 - Does this also stay the same when the Multiselect is filterable?
- [x] Provide a best practice/suggestion on what to do here.

----

### Package

carbon-components

### Package version

1.52.0

### React version

18.2.0

### Description

When focusing through the menu options of the [Combo box](https://arc.net/l/quote/qcrmkooz), the focus cycles from top to bottom. However, in the [Dropdown](https://react.carbondesignsystem.com/?path=/story/components-dropdown--default), the focus does not cycle through the options and stops at the last option in the menu.

![Apr-01-2024 10-35-36](https://github.com/carbon-design-system/carbon/assets/158570656/96579ff1-ccdd-4116-85b0-383bf8d34778)

### Reproduction/example

.

### Steps to reproduce

.

### Suggested Severity

Severity 4 = Unrelated to a user task, has a workaround or does not need a workaround.

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by tay1orjones** [comment]
Combobox and FilterableMultiSlect wraps -- MultiSelect and Dropdown do not wrap. Is this correct? The APG guidance for Combobox suggests that it should not wrap. APG for ListBox isn't clear on either approach. What should we do?

* https://www.w3.org/WAI/ARIA/apg/patterns/combobox/
* https://www.w3.org/WAI/ARIA/apg/patterns/listbox/

**Comment by mbgower** [comment]
To some degree, I feel like this is a UX discussion. There must be some decent guidance out there on why and why not one would have wrap from bottom to top item in a list. I'd like to see those considerations itemized and then ponder whether there are accessibility complications.
My basic assumption is that in a situation where all items are equally weighted, a wrap might make sense. I'm looking through a long list of dropdown items, and I want to return to a first item, having reviewed the set, so it's much easer to keep arrowing down. Think, for example of a list of movies that are available to watch.
But given that the most established dropdown component, Select, does NOT wrap around, there is clear precedent to restrict.
I think there is probably less benefit with an alphabetically organized list in one circumstances (i.e., jumping from the last Z entry to the first A) BUT if I know the item I want is near the bottom of the list, it is way faster to arrow UP from the first item and rapidly get to the last item. Of course in most interactions, I can also type any letter to jump to the first item beginning with it, so wrap has less relative utility.
I can see a user with low vision not even realizing they have wrapped. I can see some people with cognitive issues also being confused. Those are probably the biggest negatives I can think of.

**Comment by thyhmdo** [comment]
The design team will need to do more research to provide options here.

**Comment by laurenmrice** [comment]
I’ve looked into this issue, and I would lean against wrapping focus by default in these situations. Michael raised some good points around accessibility concerns, and I agree with them. There also seem to be too many conditional scenarios, where wrapping might help in some menu configurations but hinder in others, making it difficult to reliably detect/enforce the correct behavior across all use cases. I also don’t feel like wrapping gives users much of a benefit in general, so keeping it linear seems like a safer, more consistent approach.

The action item here would be for dev to implement no focus wrapping in Combobox.

**Comment by chrisgrimley** [comment]
Hi @laurenmrice. I'm just reviewing some old tickets that have the label 'needs accessibility info'. It looks like the accessibility questions have been resolved for this one. Is it ok to remove the label?

**Comment by laurenmrice** [comment]
@chrisgrimley Oh yes, we can do that.
